libraries/AP_Soaring/POMDP_Soar.cpp

```diff
@@ -0,0 +1,634 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. 
+// Licensed under the GPLv3 license
+
+#include <GCS_MAVLink/GCS.h>
+#include "AP_Soaring.h"
+#include "POMDP_Soar.h"
+
+// ArduSoar parameters
+const AP_Param::GroupInfo POMDSoarAlgorithm::var_info[] = {
+    // @Param: ENABLE
+    // @DisplayName: Is the POMDSoar algorithm on?
+    // @Description: If 1, the soaring controller uses the POMDSoar algorithm. If 0, the soaring controller uses the ArduSoar algorithm.
+    // @Units: boolean
+    // @Range: 0 1
+    // @User: Advanced
+    AP_GROUPINFO_FLAGS("ENABLE", 1, POMDSoarAlgorithm, pomdp_on, 0, AP_PARAM_FLAG_ENABLE),
+
+    // @Param: N
+    // @DisplayName: Number of samples per action trajectory used by POMDSoar
+    // @Description: Number of samples per action trajectory used by POMDSoar.
+    // @Units: samples
+    // @Range: 0 100
+    // @User: Advanced
+    AP_GROUPINFO("N", 2, POMDSoarAlgorithm, pomdp_n, 10),
+
+    // @Param: K
+    // @DisplayName: Number of POMDP sample points per 1 second of an action's trajectory used by POMDSoar.
+    // @Description: Number of POMDP sample points per 1 second of an action's trajectory used by POMDSoar.
+    // @Units: samples
+    // @Range: 0 100
+    // @User: Advanced
+    AP_GROUPINFO("K", 3, POMDSoarAlgorithm, pomdp_k, 5),
+
+    // @Param: HORI
+    // @DisplayName: POMDP planning horizon used by POMDSoar.
+    // @Description: POMDP planning horizon used by POMDSoar.
+    // @Units: seconds
+    // @Range: 0 1000
+    // @User: Advanced
+    AP_GROUPINFO("HORI", 4, POMDSoarAlgorithm, pomdp_hori, 4.0),
+
+    // @Param: STEP_T
+    // @DisplayName:POMDP planning step solve time
+    // @Description: The amount of computation time the POMDP solver has for computing the next action
+    // @Units: seconds
+    // @Range: 0 1000
+    // @User: Advanced
+    AP_GROUPINFO("STEP", 5, POMDSoarAlgorithm, pomdp_step_t, 1),
+
+    // @Param: LOOP
+    // @DisplayName: Number of POMDP solver's inner loop executions per planning step
+    // @Description: Number of POMDP solver's inner loop executions per planning step (see also the STEP_T parameter)
+    // @Units:
+    // @Range: 0 1000
+    // @User: Advanced
+    AP_GROUPINFO("LOOP", 6, POMDSoarAlgorithm, pomdp_loop_load, 1),
+
+    // @Param: ROLL1
+    // @DisplayName: POMDP's maximum commanded roll angle.
+    // @Description: Maximum commanded roll angle in the POMDP used by POMDSoar.
+    // @Units: degrees
+    // @Range: 0 1000
+    // @User: Advanced
+    AP_GROUPINFO("ROLL1", 7, POMDSoarAlgorithm, pomdp_roll1, 15),
```

Comment:
why are these needed? Why are the existing params used? If anything, they should be RLLMIN and RLLMAX. The 1 and 2 is not intuitive. Same with RRATE, might as well be RLLRT to match the others so it's intuitive

Reply:
Agreed, the existing params should be used.

Reply:
Changed the names - some can be removed by reusing existing params e.g. RLL2SRV_RMAX, LIM_ROLL_CD etc.